File: project_java/src/main/java/hu/bme/aut/onlab/dxhjoh/couchsurfing_java_backend/dto/UserDto.java
package hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.dto;

import lombok.Data;

import java.time.LocalDate;
import java.util.Set;

@Data
public class UserDto {

    private int id;
    private String fullName;
    private String username;
    private LocalDate birthDate;
    private String email;
    private String phone;
    private String password;
    private Set<RoomDto> rooms;
    private Set<BookingDto> bookings;
    private LeaderboardDto leaderboardPlace;
}

File: project/src/main/kotlin/hu/bme/aut/onlab/couchsurfing_backend/dto/UserDto.kt
package hu.bme.aut.onlab.couchsurfing_backend.dto

import lombok.Data
import java.time.LocalDate

@Data
class UserDto(val id: Int, val fullName: String, val username: String?, birthDate: LocalDate,
              email: String, phone: String, password: String) {
}